torch/csrc/inductor/aoti_eager/kernel_meta_info.cpp

```diff
@@ -25,70 +35,107 @@ TensorMetadata::TensorMetadata(
+    : is_symbolic_(is_symbolic),
+      dtype_(dtype),
+      device_(device),
+      dispatch_key_set_(dispatch_key_set),
+      sizes_(sizes),
+      strides_(strides),
+      dim_order_(dim_order),
+      requires_grad_(requires_grad) {}
+
 void TensorMetadata::build_guard(const torch::dynamo::LocalState& local_state) {
-  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(
-      !is_symbolic_, "Not support symbolic shape now");
-  std::vector<std::optional<c10::SymInt>> sym_sizes;
-  std::vector<std::optional<c10::SymInt>> sym_strides;
-  std::transform(
-      sizes_.begin(),
-      sizes_.end(),
-      std::back_inserter(sym_sizes),
-      [](int64_t size) { return std::optional<c10::SymInt>(size); });
-  std::transform(
-      strides_.begin(),
-      strides_.end(),
-      std::back_inserter(sym_strides),
-      [](int64_t stride) { return std::optional<c10::SymInt>(stride); });
   tensor_check_ = torch::dynamo::TensorCheck(
       local_state,
       nullptr,
       dispatch_key_set_,
       dtype_,
       device_.index(),
       requires_grad_,
-      sym_sizes,
-      sym_strides);
+      sizes_,
+      strides_);
 }
 
 bool TensorMetadata::operator==(const TensorMetadata& other) const {
-  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(
-      !is_symbolic_, "Not support symbolic shape now");
-
-  if (tensor_check_.has_value()) {
-    auto sizes_ = c10::IntArrayRef(other.sizes_);
-    auto strides_ = c10::IntArrayRef(other.strides_);
-    auto sym_sizes = c10::SymIntArrayRef(
-        reinterpret_cast<const c10::SymInt*>(sizes_.data()), sizes_.size());
-    auto sym_strides = c10::SymIntArrayRef(
-        reinterpret_cast<const c10::SymInt*>(strides_.data()), strides_.size());
-
-    torch::dynamo::LocalState local_state;
-    local_state.overrideDispatchKeySet(dispatch_key_set_);
-    auto _tensor_check = tensor_check_.value();
-    auto res = _tensor_check.check(
-        local_state,
-        other.dispatch_key_set_,
-        other.dtype_,
-        other.device_,
-        sym_sizes,
-        sym_strides,
-        other.requires_grad_ /* Should we need to care about grad requirement?*/);
+  TORCH_INTERNAL_ASSERT(tensor_check_.has_value());
+  torch::dynamo::LocalState local_state;
+  local_state.overrideDispatchKeySet(dispatch_key_set_);
+
+  auto _sym_sizes = std::vector<c10::SymInt>();
+  auto _sym_strides = std::vector<c10::SymInt>();
+  for (auto sym_size_val : other.sizes_) {
+    TORCH_INTERNAL_ASSERT(sym_size_val.has_value());
+    _sym_sizes.push_back(sym_size_val.value());
+  }
+  for (auto sym_stride_val : other.strides_) {
+    TORCH_INTERNAL_ASSERT(sym_stride_val.has_value());
+    _sym_strides.push_back(sym_stride_val.value());
+  }
+
+  auto sym_sizes = c10::SymIntArrayRef(
+      reinterpret_cast<const c10::SymInt*>(_sym_sizes.data()),
+      _sym_sizes.size());
+  auto sym_strides = c10::SymIntArrayRef(
+      reinterpret_cast<const c10::SymInt*>(_sym_strides.data()),
+      _sym_strides.size());
+
+  auto _tensor_check = tensor_check_.value();
```

Comment:
Originally we check `tensor_check_` has value before use. Now we don't have to?

Reply:
Yes. For now, we always ensure the `tensor_check_` is available.